### ISSUE TITLE
Wrong logo placement in Gutenberg full site template editor 

### ISSUE THREAD

**Issue by tkapler** [WordPress/gutenberg#50619]
### Description

If you compare site logo in top left corner of  post editor and site-editor
https://ai.kapler.cz/wp-admin/post.php?post=24&action=edit 
and
https://ai.kapler.cz/wp-admin/site-editor.php?canvas=edit
there is a clear difference, as the site editor one use 100 % of the area and got some ugly gray background. The difference is clear from the css

`
.edit-post-fullscreen-mode-close.components-button .edit-post-fullscreen-mode-close_site-icon {
    border-radius: 2px;
    height: 36px;
    margin-top: -1px;
    -o-object-fit: cover;
    object-fit: cover;
    width: 36px;
}
`

`
.edit-site-site-icon__image {
    background: #333;
    border-radius: 4px;
    height: auto;
    -o-object-fit: cover;
    object-fit: cover;
    width: 100%;
}
`

I hate that site editor version. It should be standardised IMO

### Step-by-step reproduction instructions

compare site icon (if you have it defined) in post editor and site editor

### Screenshots, screen recording, code snippet

![Snímek obrazovky 2023-05-14 160223](https://github.com/WordPress/gutenberg/assets/1409898/49f008be-7d11-424e-a331-d28b943f44f7)

### Environment info

- latest wp, any theme, any browser

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by Mamaduka** [comment]
I can reproduce the issue; it's probably a recent regression in the plugin.

I agree that design should be standardized across editors.

@jasmussen, @jameskoster, can you confirm the latest design so I can update one of the editors?

**Comment by jasmussen** [comment]
Yes indeed, this is partially intentional though I recognize that it's causing a bit of confusion, and could be worth a revisit. Before I explain my take on the current heuristic, GIFs of post and site editor hover animations. Post editor:

![post editor](https://github.com/WordPress/gutenberg/assets/1204802/e860b41a-3758-4961-b4bb-7018abefb830)

Site editor:

![site editor](https://github.com/WordPress/gutenberg/assets/1204802/dd6bd3e3-1922-49be-94f8-1d09f9551c5c)

The latter is the more intentional and polished one, as it's the one that received attention most recently. The idea of showing it full-bleed in the fullscreen editor is to differentiate between the edit view (fullscreen) and the site view (not fullscreen). Since it's serving as a back button, that helps situate you.

The main difference here is that the post editor at the moment has only an edit view (fullscreen), and pressing the back button takes you to the posts list. I imagine that's why it wasn't ported over.

That said, I wouldn't personally mind bringing the site editor behavior over to the post editor. 

That said, fixing https://github.com/WordPress/gutenberg/issues/49501 would also be an improvement.

**Comment by tkapler** [comment]
In my opinion, the later is "wrong". Yes, it looks nice and polished on your fake example with some painted picture but this is not how site-logo is used - it is actually a website favicon and website favicon on all the websites is something what looks nice on 16×16 and therefore everywhere it is displayed with some padding because everyone is using maximum of that given area for the symbol. Try to give there any real world favicon / site logo and you will see how ugly it looks.

**Comment by afercia** [comment]
> this is partially intentional though I recognize that it's causing a bit of confusion, and could be worth a revisit. Before I explain my take on the current heuristic, GIFs of post and site editor hover animations.

Looking a bit into this after a while. Current situation on latest trunk:
- With the default WP logo i don't see any hover animation.
- In the Site Editor the whole animation of the sidebar has changed.
- With a custom site icon, I do see a hover animation in the _post_ editor but I would argue it doesn't have any purpose in the post editor. I'd vote for removing the animation.
- In the Site editor there is no hover animation. The animation actually happens when switching View / Edit mode. In my opinion the site icon becomes too big though. To be more accurate, in View mode the site icon is smaller than the one in the post editor. In edit mode it becomes bigger than the one in the post editor. If keeping the animation is desirable, I'd vote for a maximum size that equals the size in the post editor.
- IN any case, all animations should respect the 'reduced motion' preference of the operating system. As such, I'm goin gto add the Accessibility label.

Personally, what I really don't like is that with the default WP logo there's a very evident size difference that doesn't serve any apparent purpose. See attached animated GIF below to illustrate the difference between the two editors. As an aside, the top bar total height is different by 1 pixel.

![header](https://github.com/WordPress/gutenberg/assets/1682452/36bcbecc-d242-4b38-af01-5f4e9e4c954a)

> though I recognize that it's causing a bit of confusion

Overall, I'd agree this is confusing for users and it should be improved.
